File: src/pocketcloud/cloud/network/packet/impl/normal/ProxyRegisterServerPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;

class ProxyRegisterServerPacket extends CloudPacket {

    public function __construct(
        private string $serverName = "",
        private int $port = 0
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->serverName);
        $packetData->write($this->port);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->serverName = $packetData->readString();
        $this->port = $packetData->readInt();
    }

    public function getServerName(): string {
        return $this->serverName;
    }

    public function getPort(): int {
        return $this->port;
    }

    public function handle(ServerClient $client): void {}

    public static function create(string $server, int $port): self {
        return new self($server, $port);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/ProxyRegisterServerPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.component.server.ICloudServer;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ProxyRegisterServerPacket extends CloudPacket implements ClientboundPacket {

    private String serverName;
    private String address;
    private int port;

    public ProxyRegisterServerPacket(String serverName, String address, int port) {
        this.serverName = serverName != null ? serverName : "";
        this.address = address;
        this.port = port;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(serverName, address, port);
    }

    @Override
    public void decodePayload(IPacketData packetData) {
        serverName = packetData.readString();
        address = packetData.readString();
        port = packetData.readInt();
    }

    public static ProxyRegisterServerPacket create(String serverName, String address, int port) {
        return new ProxyRegisterServerPacket(serverName, address, port);
    }

    public static ProxyRegisterServerPacket create(ICloudServer server) {
        return create(server.name(), server.data().address(), server.data().port());
    }
}
